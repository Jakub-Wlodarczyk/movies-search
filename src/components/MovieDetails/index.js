import React from 'react';
import noImage from '../../assets/sample_poster.jpg';

/**
 * The function displays some details about a movie
 * @param {Object} details
 * @param {String} baseUrl
 * @returns {JSX}
 */
const MovieDetails = ({ details, baseUrl }) => {
	const {
		budget,
		title,
		overview,
		release_date,
		revenue,
		runtime,
		poster_path
	} = details;

	return (
		<div className="movie-details">
			<div className="movie-details__movie">
				<div className="movie-details__new-search">
					<a href="/">SEARCH AGAIN</a>
				</div>
				<h1>{title}</h1>
				<p><b>BUDGET: </b>{budget} $</p>
				<p><b>REVENUE: </b>{revenue} $</p>
				<p><b>RELEASE DATE: </b>{release_date}</p>
				<p><b>RUNTIME: </b>{runtime} min</p>
				<img
					name={`${title}-image`}
					src={poster_path === undefined ? noImage : `${baseUrl}original/${poster_path}`}
				/>
				<p>{overview}</p>
			</div>
		</div>
	)
};

MovieDetails.propTypes = {
	details: React.PropTypes.shape({
		budget: React.PropTypes.number,
		title: React.PropTypes.string,
		overview: React.PropTypes.string,
		release_date: React.PropTypes.string,
		revenue: React.PropTypes.number,
		runtime: React.PropTypes.number,
		poster_path: React.PropTypes.string
	}).isRequired,
	baseUrl: React.PropTypes.string.isRequired
};

export default MovieDetails;
