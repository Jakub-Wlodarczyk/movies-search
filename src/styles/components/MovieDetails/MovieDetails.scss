.movie-details {
  display: flex;
  width: 74%;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  margin: auto;

  img {
    width: 100%;
    height: auto;
  }

  .movie-details__new-search {
    position: absolute;
    width: 200px;
    height: 200px;
    left: 0;
    top: 0;
    margin-top: 50px;
  }
}
