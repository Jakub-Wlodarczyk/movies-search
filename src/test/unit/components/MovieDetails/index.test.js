import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MovieDetails from '../../../../components/MovieDetails';

describe('MovieDetails', () => {
	it('should render the MovieDetails view', () => {
		const props = {
			details: {
				budget: 100,
				title: 'some title',
				overview: 'some longer description',
				release_date: '10-03-2005',
				revenue: 200,
				runtime: 150,
				poster_path: '/some-path'
			},
			baseUrl: '/some-base-url'
		};

		const wrapper = shallow(<MovieDetails {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render the MovieDetails container', () => {
		const props = {
			details: {
				budget: 100,
				title: 'some title',
				overview: 'some longer description',
				release_date: '10-03-2005',
				revenue: 200,
				runtime: 150,
				poster_path: '/some-path'
			},
			baseUrl: '/some-base-url'
		};

		const wrapper = shallow(<MovieDetails {...props} />);
		expect(wrapper.find('.movie-details').exists()).toBeTruthy();
	});
});
