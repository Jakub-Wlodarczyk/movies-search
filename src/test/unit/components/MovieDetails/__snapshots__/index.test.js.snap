// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render the MovieDetails view 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details__movie"
  >
    <div
      className="movie-details__new-search"
    >
      <a
        href="/"
      >
        SEARCH AGAIN
      </a>
    </div>
    <h1>
      some title
    </h1>
    <p>
      <b>
        BUDGET: 
      </b>
      100
       $
    </p>
    <p>
      <b>
        REVENUE: 
      </b>
      200
       $
    </p>
    <p>
      <b>
        RELEASE DATE: 
      </b>
      10-03-2005
    </p>
    <p>
      <b>
        RUNTIME: 
      </b>
      150
       min
    </p>
    <img
      name="some title-image"
      src="/some-base-urloriginal//some-path"
    />
    <p>
      some longer description
    </p>
  </div>
</div>
`;
